src/eval/nnue/evaluate_nnue.cpp

```diff
@@ -243,8 +243,8 @@ void load_eval() {
       return;
   }
 
-  const std::string dir_name = Options["EvalDir"];
-  const std::string file_name = Path::Combine(dir_name, NNUE::kFileName);
+  const std::string file_name = Options["EvalFile"];
+  NNUE::kFileName = file_name.c_str();
```

Comment:
The type of NNUE::kFileName is "const char*".  It holds a temporary pointer for file_name above, and will be invalid soon. Could you please change the type of NNUE::kFileName to std::string to hold the value?